### PR TITLE
[Design2-186] DSCO Tags - replace react-bootstrap tooltips with DSCO tooltips, additional updates

### PR DESCRIPTION
**Pull request by levadadenys**
### [Design2-186] DSCO Tags - replace react-bootstrap tooltips with DSCO tooltips, additional updates

- refactored `Tags` tp use DSCO `Tooltip` instead of react-bootstrap `Tooltip`
- removed unneeded styles
- updated color variables to use `primitiveColors.css`
- rewritten `TagsTest` in typescript
- refactored/optimized and added more tests to `TagsTest.tsx`

Before:

https://github.com/user-attachments/assets/07983534-9c5b-49ee-8efc-d264d3b79207

After:

https://github.com/user-attachments/assets/6dd42bfc-2b4d-4c38-8fe1-39631f49db7a

## Links
- jira ticket: [Design2-186](https://codedotorg.atlassian.net/browse/DESIGN2-186)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/componentLibrary/tags/_Tag.tsx [hunk 1]
```diff
@@ -60,28 +40,23 @@ const Tag: React.FunctionComponent<TagProps> = ({
...
-      overlay={(props: TooltipProps) => (
-        <LabelTooltip
-          id={tooltipId}
-          className={moduleStyles.tagTooltip}
-          {...props}
-        >
-          {tooltipContent}
-        </LabelTooltip>
-      )}
+    <WithTooltip
```
Comment (sanchitmalhotra126) [inline apps/src/componentLibrary/tags/_Tag.tsx:43]: 🎉

**Review by sanchitmalhotra126 — APPROVED** [review]

**Comment by bethanyaconnor** [comment]
I pulled this locally for the one place I know that we use Tags and seeing an issue. We use this on the curriculum catalog (studio.code.org/catalog) and the tooltip for the "+N" tooltips don't look right. If it helps, they're used in https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/curriculumCatalog/CardLabels.jsx.

Screenshot 2024-12-18 at 8 19 17 AM

**Comment by levadadenys** [comment]
> I pulled this locally for the one place I know that we use Tags and seeing an issue. We use this on the curriculum catalog (studio.code.org/catalog) and the tooltip for the "+N" tooltips don't look right. If it helps, they're used in https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/curriculumCatalog/CardLabels.jsx.

Thanks a lot for checking and sharing!
Gotcha, I know how to fix this. Is that ok text will be in 1 line instead of multiple lines (like on the after video in the pr description).

For the context in case someone will be looking into it in the future: reason of this bug is passing custom jsx into tooltip content instead of passing it as a string or not setting custom styles to custom jsx (it's possible that there were those styles previously, but after switching to dsco tooltip they stopped working since it now needs different css selectors.)

**Comment by bethanyaconnor** [comment]
> Is that ok text will be in 1 line instead of multiple lines (like on the after video in the pr description).

That's fine with me. Tagging @moshebaricdo in case he has a strong opinion :) 

Just because I may be OOO by the time this is fixed, as long as the tooltip looks reasonable on the catalog, I'm happy for this PR to be merged.

**Comment by moshebaricdo** [comment]
> > Is that ok text will be in 1 line instead of multiple lines (like on the after video in the pr description).
> 
> 
> 
> That's fine with me. Tagging @moshebaricdo in case he has a strong opinion :) 
> 
> 
> 
> Just because I may be OOO by the time this is fixed, as long as the tooltip looks reasonable on the catalog, I'm happy for this PR to be merged.

Works for me! Thank you for the tag (pun intended) @bethanyaconnor !

**Comment by levadadenys** [comment]
@bethanyaconnor @moshebaricdo fixed the course catalog tags usage, here's the video


https://github.com/user-attachments/assets/d1559627-964e-4a91-a2a4-14066da698fe

**Review by bethanyaconnor — APPROVED** [review]

**Pull request merged**